integration-tests/test_client_options.py

```diff
@@ -87,3 +88,21 @@ def test_support(insights_client):
     assert "Running command:" in support_result.stdout
     assert "Process output:" in support_result.stdout
     assert "Support information collected in" in support_result.stdout
+
+
+def test_client_validate_no_network_call(insights_client):
+    """Verify that '--validate' option will not try to connect to any network service.
+
+    Ref : https://bugzilla.redhat.com/show_bug.cgi?id=2009864
+    """
+
+    # '/etc/insights-client/tags.yaml' file exists with any content
+    tags_filename = "/etc/insights-client/tags.yaml"
+    subprocess.run(["touch", tags_filename])
+    validate_result = insights_client.run("--validate")
+
+    # validating tags.yaml is loaded and no metric data in output
+    assert (
+        "/etc/insights-client/tags.yaml loaded successfully" in validate_result.stdout
+    )
+    assert "metrics Metrics:" not in validate_result.stdout
```

Comment:
checking that the removed message is not present seems a good starting point; however, we cannot be sure that something else is not actually doing any network communication

I'd suggest something else instead, based on what done in #213 (for `test_wrong_url_connection`)
- disable `auto_update` & `auto_config`
- set a non-existing `base_url`
- run `insights-client --validate`, and expect it to work

this way, if anything attempts any communication to the Insights server, `insights-client` will fail

Reply:
I agree mostly but this is running when insights-client is not registered, if anything that needs network communication during --validate should fail even if base_url is correct.

Reply:
done